lib/cache/collections.go

```diff
@@ -1246,9 +1246,20 @@ func (databaseExecutor) getAll(ctx context.Context, cache *Cache, loadSecrets bo
 
 func (databaseExecutor) upsert(ctx context.Context, cache *Cache, resource types.Database) error {
 	if err := cache.databasesCache.CreateDatabase(ctx, resource); err != nil {
+		// Skip databases that failed to be validated to avoid stopping cache
+		// backend events consumption (which can lead to an invalid cache
+		// state). This case can happen during Teleport upgrades/downgrades,
+		// where the database stored on the backend is no longer valid given the
+		// Teleport version.
```

Comment:
> This case can happen during Teleport upgrades/downgrades, where the database stored on the backend is no longer valid given the Teleport version.

That is a _significant_ backwards compatibility bug that we should fix and prevent in the future, instead of fixing the symptom of a bad cache in a way that can lead to really weird behavior (such as `Create` failing while `Get` returns nothing - but only if the auth cache itself is healthy, otherwise `Get` will error out).